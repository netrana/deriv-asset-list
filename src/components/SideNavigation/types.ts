export type NavigationMenuItem = {
  title: string;
  link: string;
  routeName?: string;
  isDisabled?: boolean;
};

export interface Props {
  menuItems?: NavigationMenuItem[];
  className?: string;
  selectedTab?: string;
  onClick: (link: string)=> void;
}
