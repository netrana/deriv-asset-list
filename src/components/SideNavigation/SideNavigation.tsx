import React, { FC } from "react";
import classNames from "classnames";

import styles from "./sideNavigation.module.scss";

import { Props } from "./types";
import Menu from "antd/lib/menu";


const Sidebar: FC<Props> = (props) => {
  const { menuItems, className = "", selectedTab = "", onClick } = props;

  return (
    <div
      className={classNames({
        [styles.sidebarContainer]: true,
        [className]: Boolean(className),
      })}
    >
      <Menu selectedKeys={[selectedTab]}>
        {/* <Menu.Item disabled>Menu1</Menu.Item> 
        <Menu.Item>Menu2</Menu.Item> 
        <Menu.Item>Menu3</Menu.Item> 
        <Menu.Item>Menu4</Menu.Item>   */}
        {menuItems && menuItems.map((menuItem, index) => (
          <Menu.Item disabled={menuItem.isDisabled} key={menuItem.title}  onClick={() => onClick(menuItem.link)}>
            {menuItem.title}
          </Menu.Item>
          // <div key={index}>
          //   <div
          //     className={classNames({
          //       [styles.title]: true,
          //       [styles.active]: menuItem.title === selectedTab,
          //     })}
          //     onClick={() => onClick(menuItem.link)}
          //   >
          //     {menuItem.title}
          //   </div>
          //   <br />
          // </div>
        ))}
      </Menu>

    </div>
  );
};

export default Sidebar;
