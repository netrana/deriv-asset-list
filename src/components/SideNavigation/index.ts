import SidebarNavigation from "./SideNavigation";

export default SidebarNavigation;