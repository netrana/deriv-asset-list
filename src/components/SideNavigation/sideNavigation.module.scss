@import 'styles/globals.scss';

.sidebarContainer {
  margin-left: 46px;
  margin-top: 46px;
  width: 200px;
  .active {
    color: $primaryThemeColor;
  }
  .title {
    cursor: pointer;
  }
  :global(.ant-menu-item-selected) {
    color: $primaryThemeColor;
    background-color: white !important;
    background-image: none !important;
  }

  :global(.ant-menu-item:hover) {
    color: $primaryThemeColor;
    background-color: white !important;
    background-image: none !important;
  }

  :global(.ant-menu-item::after) {
    border-right: 0;
  }
  :global(.ant-menu-vertical) {
    border: none !important;
  }
}
