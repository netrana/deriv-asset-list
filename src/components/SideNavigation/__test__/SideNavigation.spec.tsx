import React from "react";
import render from "react-test-renderer";

import SideNavigation from "../SideNavigation";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

describe("SideNavigation component test", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const tree = render
        .create(
          <SideNavigation
            onClick={jest.fn}
            menuItems={[
              {
                title: "test",
                link: "test",
              },
            ]}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
