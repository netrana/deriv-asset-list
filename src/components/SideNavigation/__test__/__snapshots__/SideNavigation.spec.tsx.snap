// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation component test Snapshots should match snapshot 1`] = `
<div
  className="sidebarContainer"
>
  <ul
    className="ant-menu ant-menu-root ant-menu-vertical ant-menu-light"
    data-menu-list={true}
    onKeyDown={[Function]}
    role="menu"
    tabIndex={0}
  >
    <li
      className="ant-menu-item ant-menu-item-only-child"
      data-menu-id={null}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        test
      </span>
    </li>
  </ul>
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
